data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/GenesisTimeHandler.java

```diff
@@ -13,24 +13,49 @@
 
 package tech.pegasys.artemis.beaconrestapi.beaconhandlers;
 
-import tech.pegasys.artemis.beaconrestapi.handlerinterfaces.BeaconRestApiHandler;
-import tech.pegasys.artemis.storage.ChainStorageClient;
+import static javax.servlet.http.HttpServletResponse.SC_INTERNAL_SERVER_ERROR;
 
-public class GenesisTimeHandler implements BeaconRestApiHandler {
+import com.google.common.primitives.UnsignedLong;
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+import tech.pegasys.artemis.provider.JsonProvider;
+import tech.pegasys.artemis.storage.ChainStorageClient;
 
-  private final ChainStorageClient client;
+public class GenesisTimeHandler implements Handler {
+  private final Logger LOG = LogManager.getLogger();
+  public static final String ROUTE = "/node/genesis_time/";
+  ChainStorageClient chainStorageClient;
 
-  public GenesisTimeHandler(final ChainStorageClient client) {
-    this.client = client;
-  }
-
-  @Override
-  public String getPath() {
-    return "/node/genesis_time";
+  public GenesisTimeHandler(ChainStorageClient chainStorageClient) {
+    this.chainStorageClient = chainStorageClient;
   }
 
+  @OpenApi(
+      path = GenesisTimeHandler.ROUTE,
+      method = HttpMethod.GET,
+      summary = "Get the genesis_time parameter from beacon node configuration.",
+      tags = {"Node"},
+      description =
+          "Requests the genesis_time parameter from the beacon node, which should be consistent across all beacon nodes that follow the same beacon chain.",
+      responses = {
+        @OpenApiResponse(status = "200", content = @OpenApiContent(from = UnsignedLong.class)),
+        @OpenApiResponse(status = "500")
+      })
   @Override
-  public Object handleRequest(RequestParams params) {
-    return client.getGenesisTime();
+  public void handle(@NotNull Context ctx) throws Exception {
```

Comment:
nit: We don't typically use the `@NotNull` annotations (IntelliJ does sneak them in from time to time). They are unfortunately on the class path because of Kotlin dependencies.